src/test/java/seedu/address/logic/commands/ReassignCommandTest.java

```diff
@@ -0,0 +1,201 @@
+    }
+
+    @Test
+    public void execute_overlappingBookingReassign_throwsCommandException() {
+        RoomNumber roomNumber = ROOM_NUMBER_011;
+        LocalDate startDate = TODAY_TOMORROW.getStartDate();
+        RoomNumber newRoomNumber = ROOM_NUMBER_020;
+        ReassignCommand reassignCommand = new ReassignCommand(roomNumber, startDate, newRoomNumber);
+
+        String expectedMessage = String.format(ReassignCommand.MESSAGE_OVERLAPPING_BOOKING, newRoomNumber);
+
+        assertCommandFailure(reassignCommand, model, commandHistory, expectedMessage);
+    }
+
+    @Test
+    public void execute_validReassign_success() {
+        RoomNumber roomNumber = ROOM_NUMBER_011;
+        LocalDate startDate = TODAY_TOMORROW.getStartDate();
+        RoomNumber newRoomNumber = ROOM_NUMBER_031;
+        ReassignCommand reassignCommand = new ReassignCommand(roomNumber, startDate, newRoomNumber);
+
+        String expectedMessage = String.format(ReassignCommand.MESSAGE_REASSIGN_SUCCESS,
+                ParserUtil.parseDateToString(startDate), roomNumber, newRoomNumber);
+
+        Model expectedModel = new ModelManager(model.getConcierge(), new UserPrefs());
+        expectedModel.reassignRoom(roomNumber, startDate, newRoomNumber);
+        expectedModel.commitConcierge();
+
+        assertCommandSuccess(reassignCommand, model, commandHistory, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_validCheckedInReassign_success() {
+        RoomNumber roomNumber = ROOM_NUMBER_012;
+        LocalDate startDate = TODAY_TOMORROW.getStartDate();
+        RoomNumber newRoomNumber = ROOM_NUMBER_031;
+        ReassignCommand reassignCommand = new ReassignCommand(roomNumber, startDate, newRoomNumber);
+
+        String expectedMessage = String.format(ReassignCommand.MESSAGE_REASSIGN_SUCCESS,
+                ParserUtil.parseDateToString(startDate), roomNumber, newRoomNumber);
+
+        Model expectedModel = new ModelManager(model.getConcierge(), new UserPrefs());
+        expectedModel.reassignRoom(roomNumber, startDate, newRoomNumber);
+        expectedModel.commitConcierge();
+
+        assertCommandSuccess(reassignCommand, model, commandHistory, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void executeUndoRedo_validReassign_success() throws Exception {
+        RoomNumber roomNumber = ROOM_NUMBER_012;
+        LocalDate startDate = TODAY_TOMORROW.getStartDate();
+        RoomNumber newRoomNumber = ROOM_NUMBER_031;
+        ReassignCommand reassignCommand = new ReassignCommand(roomNumber, startDate, newRoomNumber);
+
+        Model expectedModel = new ModelManager(model.getConcierge(), new UserPrefs());
+        expectedModel.reassignRoom(roomNumber, startDate, newRoomNumber);
+        expectedModel.commitConcierge();
+
+        // reassign -> booking reassigned
+        reassignCommand.execute(model, commandHistory);
+
+        // undo -> reverts concierge back to previous state
```

Comment:
Hmm, would this be a system test instead of a unit test?

Reply:
Nope, it's neither. It's an integration test, as specified in the test's Javadoc.